packages/webapp/pages/squads/[handle]/[token].tsx

```diff
@@ -0,0 +1,172 @@
+import classed from '@dailydotdev/shared/src/lib/classed';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import Link from 'next/link';
+import SourceButton from '@dailydotdev/shared/src/components/cards/SourceButton';
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import React, { ReactElement } from 'react';
+import { ParsedUrlQuery } from 'querystring';
+import {
+  GetStaticPathsResult,
+  GetStaticPropsContext,
+  GetStaticPropsResult,
+} from 'next';
+import { webappUrl } from '@dailydotdev/shared/src/lib/constants';
+import { NextSeo } from 'next-seo';
+import { getLayout } from '../../../components/layouts/MainLayout';
+
+const getOthers = (others: Edge<SquadMember>[]) => {
+  const { length } = others;
+  if (length === 0) {
+    return 'is';
+  }
+
+  if (length === 1) {
+    const member = others[0].node.user.name;
+    return `and ${member} are`;
+  }
+
+  return `and ${length} others are`;
+};
+
+const BodyParagraph = classed('p', 'typo-body text-theme-label-tertiary');
+const HighlightedText = classed('span', 'font-bold text-theme-label-primary');
+
+export interface SquadReferralProps {
+  token: string;
+  handle: string;
+}
+
+const SquadReferral = ({ token, handle }: SquadReferralProps): ReactElement => {
+  const router = useRouter();
+  const { showLogin, user: loggedUser } = useAuthContext();
+  const { data: member } = useQuery(
+    ['squad_referral', token, loggedUser?.id],
+    () => getSquadInvitation(token),
+    {
+      enabled: !!token,
+      onSuccess: (response) => {
+        if (!loggedUser) return null;
+
+        if (!response?.source?.id) return router.replace(webappUrl);
+
+        const sourceId = response.source.id;
+        const squadsUrl = `/squads/${sourceId}`;
+        const isValid = validateSourceId(handle, response.source);
+
+        if (!isValid) return router.replace(webappUrl);
+
+        const isMember = response.source.members.edges.some(
+          ({ node }) => node.user.id === loggedUser.id,
+        );
+        if (isMember) return router.replace(squadsUrl);
+
+        return null;
+      },
+    },
+  );
+  const sourceId = member?.source?.id;
+  const squadsUrl = `/squads/${sourceId}`;
+  const { mutateAsync: onJoinSquad } = useMutation(
+    () => joinSquadInvitation({ sourceId, token }),
+    { onSuccess: () => router.replace(squadsUrl) },
+  );
+
+  const onJoinClick = async () => {
+    if (!loggedUser) return showLogin('join squad');
+
+    return onJoinSquad();
+  };
+
+  if (!member) {
+    return null;
+  }
+
+  const { user, source } = member;
+  const others = source.members.edges.filter(
+    ({ node }) => node.user.id !== user.id,
+  );
+  const othersLabel = getOthers(
+    source.members.edges.filter(({ node }) => node.user.id !== user.id),
+  );
+
+  return (
+    <PageContainer className="relative justify-center items-center min-h-[calc(100vh-3.5rem)]">
```

Comment:
Is the min height calc really needed?
Seems like we don't use it anywhere else at the moment

Reply:
I have used padding instead of min height. WDYT?

Reply:
Yep! Nice one